File: src/main/java/com/radar/prospection/scraper/MissionScraper.java
package com.radar.prospection.scraper;

import com.radar.prospection.domain.Mission;
import com.radar.prospection.domain.Source;

import java.util.List;

public interface MissionScraper {

    Source getSource();

    List<Mission> scrape(List<String> keywords);

    default boolean supports(Source source) {
        return getSource() == source;
    }
}

File: backend/src/main/kotlin/com/radar/prospection/scraper/MissionScraper.kt
package com.radar.prospection.scraper

import com.radar.prospection.domain.Mission
import com.radar.prospection.domain.Source

interface MissionScraper {
    fun getSource(): Source
    fun scrape(keywords: List<String>): List<Mission>
    fun supports(source: Source): Boolean = getSource() == source
}
